SignalMessaging/utils/ConversationStyle.swift

```diff
@@ -136,76 +136,131 @@ public class ConversationStyle: NSObject {
 
     private static let defaultBubbleColorIncoming = UIColor.ows_messageBubbleLightGray
 
-    // TODO:
     @objc
-    public let bubbleColorOutgoingUnsent = UIColor.ows_red
+    public let bubbleColorOutgoingUnsent = UIColor.gray
 
-    // TODO:
     @objc
-    public let bubbleColorOutgoingSending = UIColor.ows_light35
+    public let bubbleColorOutgoingSending = UIColor.ows_fadedBlue
 
     @objc
-    public let bubbleColorOutgoingSent = UIColor.ows_light10
+    public let bubbleColorOutgoingSent = UIColor.ows_materialBlue
 
     @objc
     public var primaryColor: UIColor
 
     @objc
     public func bubbleColor(message: TSMessage) -> UIColor {
         if message is TSIncomingMessage {
-            return primaryColor
+            return ConversationStyle.defaultBubbleColorIncoming
         } else if let outgoingMessage = message as? TSOutgoingMessage {
             switch outgoingMessage.messageState {
             case .failed:
-                return self.bubbleColorOutgoingUnsent
+                return bubbleColorOutgoingUnsent
             case .sending:
-                return self.bubbleColorOutgoingSending
+                return bubbleColorOutgoingSending
             default:
-                return self.bubbleColorOutgoingSent
+                return bubbleColorOutgoingSent
             }
         } else {
             owsFail("Unexpected message type: \(message)")
-            return UIColor.ows_materialBlue
+            return bubbleColorOutgoingSent
         }
     }
 
     @objc
     public func bubbleColor(call: TSCall) -> UIColor {
-        if call.isIncoming {
-            return primaryColor
+        return bubbleColor(isIncoming: call.isIncoming)
+    }
+
+    @objc
+    public func bubbleColor(isIncoming: Bool) -> UIColor {
+        if isIncoming {
+            return ConversationStyle.defaultBubbleColorIncoming
         } else {
             return self.bubbleColorOutgoingSent
         }
     }
 
     @objc
-    public static var bubbleTextColorIncoming = UIColor.ows_white
+    public static var bubbleTextColorIncoming = UIColor.ows_black
+    public static var bubbleTextColorOutgoing = UIColor.ows_white
 
     @objc
     public func bubbleTextColor(message: TSMessage) -> UIColor {
         if message is TSIncomingMessage {
             return ConversationStyle.bubbleTextColorIncoming
-        } else if let outgoingMessage = message as? TSOutgoingMessage {
-            switch outgoingMessage.messageState {
-            case .failed:
-                return UIColor.ows_black
-            case .sending:
-                return UIColor.ows_black
-            default:
-                return UIColor.ows_black
-            }
+        } else if let _ = message as? TSOutgoingMessage {
```

Comment:
If you don't care to use the message as a TSOutgoingMessage, and just want to type check, consider using:

```
} else if message is TSOutgoingMessage {
```

Or maybe a switch...

Take it or leave it of course.

Reply:
Done.